frontend/src/components/auth/RegisterForm.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+
+import { Form, Input, Button, Alert} from 'antd';
+import { UserOutlined, LockOutlined, GithubOutlined, FontSizeOutlined } from '@ant-design/icons';
+
+// Redux
+import {connect} from 'react-redux';
+
+// Type checking of props
+import PropTypes from 'prop-types';
+
+import {Redirect} from 'react-router-dom';
+
+// Auth Action
+import {register} from '../../actions/authActions';
+import {clearErrors} from '../../actions/errorActions';
+
+
+class RegisterForm extends React.Component {
+    state = {
+        msg: null,
+    };
+
+    componentDidUpdate(prevProps) {
+        const { error, isAuthenticated} = this.props;
+        if (error !== prevProps.error) {
+            // Check for a register error
+            if (error.id === 'REGISTER_FAIL') {
+               this.setState({msg: error.msg}); 
+            } else {
+                this.setState({msg: null});
+            }
+        }
+        
+        // This should be authenticated routing
+        if (isAuthenticated) {
+            // Send the user to the homepage
+            console.log('Sending user to the homepage');
+        }
+    }
+
+    componentWillUnmount() {
```

Comment:
I am not sure if react lifecycle will follow the intended order. 

Please make sure the props are working as you expected. Using lifecycle is a good way to update dynamic data but must be careful

Reply:
yep this works because when we switch components it causes a re-render so the errors are cleared